src/databricks/labs/ucx/workspace_access/clusters.py

```diff
@@ -0,0 +1,54 @@
+import logging
+
+from databricks.labs.blueprint.tui import Prompts
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors import InvalidParameterValue
+from databricks.sdk.service.compute import DataSecurityMode
+
+logger = logging.getLogger(__name__)
+
+
+class ClusterAccess:
+    def __init__(self, prompts: Prompts, ws: WorkspaceClient):
+        self._ws = ws
+        self._prompts = prompts
+
+    @classmethod
+    def current(cls, ws: WorkspaceClient):
+        prompts = Prompts()
+        return ClusterAccess(prompts, ws)
+
+    def map_cluster_to_uc(self, cluster_id: str | None = None):
+        spark_version = self._ws.clusters.select_spark_version(latest=True)
+        security_modes = {"Single User": DataSecurityMode.SINGLE_USER, "Shared": DataSecurityMode.USER_ISOLATION}
```

Comment:
use this for automated upgrade logic: https://docs.databricks.com/en/archive/compute/cluster-ui-preview.html#how-does-backward-compatability-work-with-these-changes